src/dmd/safe.d

```diff
@@ -107,7 +107,6 @@ bool isSafeCast(Expression e, Type tfrom, Type tto)
     if (!tto.hasPointers())
         return true;
 
-    auto tfromb = tfrom.toBasetype();
     auto ttob = tto.toBasetype();
 
     if (ttob.ty == Tclass && tfrom.ty == Tclass)
```

Comment:
This might actually be a typo.

Reply:
@wilzbach - worth replacing `tfrom` with `tfromb` here?  PR that introduced was #6090

Reply:
https://github.com/dlang/dmd/blob/74285e41e670bf7e2ae7d83bc9b1617e12f93a65/src/dmd/mtype.d#L2293-L2296

https://github.com/dlang/dmd/blob/74285e41e670bf7e2ae7d83bc9b1617e12f93a65/src/dmd/mtype.d#L7966-L7972

1) It definitely was a typo
2) `toBasetype` only has an effect for `TypeEnum` 
3) The following isn't legal anyways:

```d
enum foo = new Object();
```
https://run.dlang.io/is/F0dHH5

4) We should be consistent -> let's rename `tfrom` to `tfromb` (or drop `ttob`)